### ISSUE TITLE
Style/SoleNestedConditional: autocorrect generates broken conditional statements

### ISSUE THREAD

**Issue by bcoles** [rubocop/rubocop#14177]
`rubocop -a --only Style/SoleNestedConditional` converts valid (but poorly written) code into broken code. This cop breaks code despite being considered a safe autocorrect (`-a`).

## Expected behavior

Input:

```ruby
# cat asdf
#!/usr/bin/env ruby
res = 'abc353'
if res =~ /353/
  if not res =~ /366/
    puts 'true'
  end
end
```

Auto-corrected code:

```ruby
#!/usr/bin/env ruby
res = 'abc353'
if res =~ /353/ && res !~ /366/
  puts 'true'
end
```

## Actual behavior

Input:

```ruby
# cat asdf
#!/usr/bin/env ruby
res = 'abc353'
if res =~ /353/
  if not res =~ /366/
    puts 'true'
  end
end
```

```
# ./asdf 
true
```

Output:

```
# rubocop --debug -a asdf
For /root: Default configuration from /var/lib/gems/3.3.0/gems/rubocop-1.75.5/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /root/asdf
F

Offenses:

asdf:3:26: F: Lint/Syntax: unexpected token tIDENTIFIER
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
if (res =~ /353/) && not res =~ /366/
                         ^^^
asdf:4:3: C: [Corrected] Style/IfUnlessModifier: Favor modifier if usage when having a single-line body. Another good alternative is the usage of control flow &&/||.
  if not res =~ /366/
  ^^
asdf:4:3: C: [Corrected] Style/NegatedIf: Favor unless over if for negative conditions.
  if not res =~ /366/ ...
  ^^^^^^^^^^^^^^^^^^^
asdf:4:3: C: [Corrected] Style/SoleNestedConditional: Consider merging nested conditions into outer if conditions.
  if not res =~ /366/
  ^^
asdf:4:6: C: [Corrected] Style/Not: Use ! instead of not.
  if not res =~ /366/
     ^^^
asdf:4:15: C: [Corrected] Style/SoleNestedConditional: Consider merging nested conditions into outer if conditions.
  puts 'true' if not res =~ /366/
              ^^
asdf:4:18: C: [Corrected] Style/Not: Use ! instead of not.
  puts 'true' if not res =~ /366/
                 ^^^
asdf:5:1: F: Lint/Syntax: unexpected token kEND
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
end
^^^

1 file inspected, 8 offenses detected, 6 offenses corrected
Finished in 0.3222917440580204 seconds
```

Auto-corrected code:

```ruby
# cat asdf
#!/usr/bin/env ruby
res = 'abc353'
if (res =~ /353/) && not res =~ /366/
  puts 'true'
end
```

```
# ./asdf 
./asdf: 
./asdf:3: syntax error, unexpected local variable or method, expecting '(' (SyntaxError)
if (res =~ /353/) && not res =~ /366/
                         ^~~
```

Reverting `asdf` and running the `Style/SoleNestedConditional` cop only:

```
# rubocop -a --only Style/SoleNestedConditional --debug asdf
For /root: Default configuration from /var/lib/gems/3.3.0/gems/rubocop-1.75.5/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /root/asdf
F

Offenses:

asdf:3:26: F: Lint/Syntax: unexpected token tIDENTIFIER
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
if (res =~ /353/) && not res =~ /366/
                         ^^^
asdf:4:3: C: [Corrected] Style/SoleNestedConditional: Consider merging nested conditions into outer if conditions.
  if not res =~ /366/
  ^^
asdf:5:3: F: Lint/Syntax: unexpected token kEND
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
  end
  ^^^

1 file inspected, 3 offenses detected, 1 offense corrected
Finished in 0.20269619300961494 seconds
```

Auto-corrected code:

```ruby
# cat asdf
#!/usr/bin/env ruby
res = 'abc353'
if (res =~ /353/) && not res =~ /366/
    puts 'true'
  end
```

```
# ./asdf 
./asdf: 
./asdf:3: syntax error, unexpected local variable or method, expecting '(' (SyntaxError)
if (res =~ /353/) && not res =~ /366/
                         ^~~
```

## Steps to reproduce the problem

See above.

## RuboCop version

```
# rubocop -V
1.75.5 (using Parser 3.3.8.0, rubocop-ast 1.44.1, analyzing as Ruby 2.7, running on ruby 3.3.7) [x86_64-linux-gnu]
```
